File: sixthSeminar/src/main/java/org/sopt/sixthSeminar/controller/ServiceMemberController.java
package org.sopt.sixthSeminar.controller;

import lombok.RequiredArgsConstructor;
import org.sopt.sixthSeminar.common.dto.SuccessResponse;
import org.sopt.sixthSeminar.dto.request.serviceMember.ServiceMemberRequest;
import org.sopt.sixthSeminar.dto.response.member.MemberSignInResponse;
import org.sopt.sixthSeminar.exception.SuccessMessage;
import org.sopt.sixthSeminar.service.ServiceMemberService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.net.URI;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/users/")
public class ServiceMemberController {

    private final ServiceMemberService serviceMemberService;

    @PostMapping("sign-up")
    public SuccessResponse signUp(final @RequestBody ServiceMemberRequest request) {
        return SuccessResponse.success(SuccessMessage.MEMBER_CREATE_SUCCESS);
    }

    @PostMapping("sign-in")
    public SuccessResponse<MemberSignInResponse> signIn(final @RequestBody ServiceMemberRequest request) {
        return SuccessResponse.success(SuccessMessage.MEMBER_SIGN_IN_SUCCESS,
                serviceMemberService.signIn(request));
    }
}

File: sixthKotlin/src/main/kotlin/org/sopt/sixthKotlin/controller/ServiceMemberController.kt
package org.sopt.sixthKotlin.controller

import org.sopt.sixthKotlin.controller.dto.request.ServiceMemberRequest
import org.sopt.sixthKotlin.controller.dto.response.MemberSignInResponse
import org.sopt.sixthKotlin.service.ServiceMemberService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/users/")
class ServiceMemberController(
    private val serviceMemberService: ServiceMemberService
) {
    @PostMapping("sign-up")
    fun signUp(@RequestBody request: ServiceMemberRequest): ResponseEntity<Void> {
        serviceMemberService.create(request)
        return ResponseEntity.ok().build()
    }

    @PostMapping("sign-in")
    fun signIn(@RequestBody request: ServiceMemberRequest): ResponseEntity<MemberSignInResponse> {
        return ResponseEntity.ok(serviceMemberService.signIn(request))
    }
}